src/app/js/public/search/SearchResultSort.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { field as fieldProptypes } from '../../propTypes';
+import SortButton from '../../lib/components/SortButton';
+
+const SearchResultSort = ({ fields, fieldNames, sort, sortBy, sortDir }) => {
+    const titleField = fields.find(field => field.name === fieldNames.title);
+    const descriptionField = fields.find(
+        field => field.name === fieldNames.description,
+    );
+
+    const handleSort = name => {
+        sort({ sortBy: name });
+    };
+
+    return (
+        <div>
+            {titleField && (
+                <SortButton
+                    className={`sort_${fieldNames.title}`}
+                    name={fieldNames.title}
+                    label={titleField.label}
+                    sort={() => handleSort(fieldNames.title)}
+                    sortBy={sortBy}
+                    sortDir={sortDir}
+                />
+            )}
+            {descriptionField && (
+                <SortButton
+                    className={`sort_${fieldNames.description}`}
+                    name={fieldNames.description}
+                    label={descriptionField.label}
+                    sort={() => handleSort(fieldNames.description)}
+                    sortBy={sortBy}
+                    sortDir={sortDir}
+                />
```

Comment:
code is duplicate is it possible to factorize

Reply:
We could create a loop, but is it worth it for 2 (max 4) components?